prebuilt-ui/basic-embed/components/PrebuiltCall.js

```diff
@@ -0,0 +1,254 @@
+          setDemoState('call');
+          setExp(roomExp);
+          setIsError(false);
+        } catch (e) {
+          setDemoState('home');
+          setIsError(true);
+        }
+      }
+    },
+    [roomURL, exp]
+  );
+
+  // Updates state if a room is provided
+  const handleRoomInput = useCallback(
+    (e) => {
+      setRoomURL(e?.target?.value);
+      if (e?.target?.checkValidity()) {
+        setRoomValidity(true);
+        console.log(roomValidity);
+      }
+    },
+    [roomValidity]
+  );
+
+  const submitJoinRoom = useCallback(() => {
+    setDemoState('call');
+  });
+
+  const handleCopyClick = useCallback(() => {
+    console.log('click');
+    writeText(roomURL);
+    setLinkCopied(true);
+    setTimeout(() => setLinkCopied(false), 5000);
+  }, [roomURL, linkCopied]);
+
+  const content = useMemo(() => {
+    switch (demoState) {
+      case 'home':
+        return (
+          <Card>
+            <CardHeader>
+              Start demo with a new unique room, or paste in your own room URL.
+            </CardHeader>
+            <CardBody>
+              {!props.configured && (
+                <Well variant="error">
+                  You must configure env variables to create rooms (see README
+                  instructions).
+                </Well>
+              )}
+              {isError && (
+                <Well variant="error">
+                  Error creating the room. Please try again.
+                </Well>
+              )}
+              <Button onClick={() => createRoom()} disabled={roomValidity}>
+                Create room and start
+              </Button>
+              <Field label="Or enter room to join" className="roomField">
+                <TextInput
+                  ref={roomURLRef}
+                  type="text"
+                  placeholder="Enter room URL..."
+                  pattern="^(https:\/\/)?[\w.-]+(\.(daily\.(co)))+[\/\/]+[\w.-]+$"
+                  onChange={handleRoomInput}
+                />
+              </Field>
+              <CardFooter>
+                <Button
+                  onClick={() => submitJoinRoom()}
+                  disabled={!roomValidity}
+                >
+                  Join room
+                </Button>
+              </CardFooter>
+            </CardBody>
+          </Card>
+        );
+      case 'call':
+        return (
+          <>
+            <div ref={iframeRef} className="call" />
+            <Card>
+              <CardHeader>Copy and share the URL to invite others.</CardHeader>
+              <CardBody>
+                <label for="copy-url"></label>
+                <TextInput
+                  type="text"
+                  class="url-input"
+                  id="copy-url"
+                  placeholder="Copy this room URL"
+                  value={roomURL}
+                  pattern="^(https:\/\/)?[\w.-]+(\.(daily\.(co)))+[\/\/]+[\w.-]+$"
+                />
+                <Button onClick={handleCopyClick}>
```

Comment:
this has a border and a box shadow when you click it. Feels a little funny to me (but maybe doesn't matter since this will get some design time at some point)

Reply:
Yes, I think this is okay for now since we'll override those shared components eventually! Whatever I fix in the shared component will just be undone, so I'll leave for now.